sound/soc/sof/intel/hda.c

```diff
@@ -836,7 +836,7 @@ static irqreturn_t hda_dsp_interrupt_thread(int irq, void *context)
 	if (hda_dsp_check_ipc_irq(sdev))
 		sof_ops(sdev)->irq_thread(irq, sdev);
 
-	if (hda_dsp_check_sdw_irq(sdev))
+	if (snd_sof_dsp_check_sdw_irq(sdev))
```

Comment:
We may move snd_sof_dsp_check_sdw_irq(sdev) to a better place once we find a better place for it.

Reply:
Does hda_sdw_check_wakeen_irq needs another ops for new platform ?  And can we merge ipc, sdw, wakeen irq into one ops then we don't need to have 3 ops for different irq.

Reply:
> Does hda_sdw_check_wakeen_irq need another ops for new platform ? 

@RanderWang Looking at the code [#define HDA_DSP_REG_SNDW_WAKE_STS      0x2C192](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/intel/hda.h#L236). I think it is actually [SDW_SHIM_BASE	(0x2C000)](https://github.com/thesofproject/linux/blob/topic/sof-dev/drivers/soundwire/intel_init.c#L22) + [SDW_SHIM_WAKESTS	(0x192)](https://github.com/thesofproject/linux/blob/topic/sof-dev/drivers/soundwire/intel.c#L64). We will have different sdw shim base, so some change is required. However, maybe we don't need a new ops.

>  And can we merge ipc, sdw, wakeen irq into one ops then we don't need to have 3 ops for different irq.

I would prefer to have separate ops since we have different handler. Or maybe a ops which return a flag to indicate which irq is triggered. Like 
```
#define STREAM_IRQ	BIT(0)
#define IPC_IRQ		BIT(1)
...
```
What do you think? @plbossart

Reply:
Sorry I don't understand @RanderWang's comment nor @bardliao 's answer.
We have different sources of interrupts, and we need to check for each source and deal with it as appropriate.
Maybe we can handle Cadence and wakeen interrupts together, since they are both related to SoundWire.
But I don't see the rationale for handling streams or IPCs in the same ops as SoundWire, there are multiple cases where SoundWire is disabled. We should not mix things that are not connected?

Reply:
@plbossart The new comments may solve your confusion. :)